rest_server/src/controllers/item_controller.js

```diff
@@ -27,11 +35,17 @@ function checkReadPermission(userInfo, item) {
   return false;
 }
 
-function checkWritePermission(tokenInfo, item) {
+async function checkWritePermission(tokenInfo, item, categories) {
   if (tokenInfo.admin === true) {
     return true;
   }
-  if (tokenInfo.username === item.author) {
+  if (categories === undefined) {
+    categories = await MarketplaceItem.getCategories(item);
```

Comment:
>MarketplaceItem.getCategories(item); [](http://example.com/codeflow?start=23&length=36)

is it possible that the return of getCategories is undefined?

Reply:
The getCategories will return `Promise<Array<Model>>`, I think it will not return undefined.
Reference to https://sequelize.org/master/class/lib/associations/belongs-to-many.js~BelongsToMany.html